ocaml/src/tools/alba_rora_stubs.c

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright (C) 2016 iNuron NV
+
+This file is part of Open vStorage Open Source Edition (OSE), as available from
+
+
+    http://www.openvstorage.org and
+    http://www.openvstorage.com.
+
+This file is free software; you can redistribute it and/or modify it
+under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+as published by the Free Software Foundation, in version 3 as it comes
+in the <LICENSE.txt> file of the Open vStorage OSE distribution.
+
+Open vStorage is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY of any kind.
+*/
+
+#include <caml/mlvalues.h>
+#include <caml/bigarray.h>
+#include <caml/alloc.h>
+#include <caml/memory.h>
+#include <caml/custom.h>
+#include <caml/fail.h>
+#include <caml/threads.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <rocksdb/c.h>
+#include <string.h>
+#include <assert.h>
+#include <gobjfs_server.h>
+#include <stdbool.h>
+#include <sys/stat.h>
+
+static void* _DB_POINTER = NULL;
+static char* _ROOT_PATH = NULL;
+static int   _ROOT_PATH_LENGTH = -1;
+
+
+rocksdb_t* alba_rocksdb_open_and_register(
+    const rocksdb_options_t* options,
+    const char* name, char** errptr
+    ){
+    rocksdb_t* result = rocksdb_open(options, name, errptr);
+    _DB_POINTER = result;
+    return result;
+}
+
+
+char hexlify(uint8_t c) {
+    char r;
+    if (c < 10) {
+        r = c + 48;
+    } else {
+        r = c + 87;
+    }
+    return r;
+}
+
+void convert_hex(uint8_t c, char* tgt) {
+    char h0 = hexlify(c >> 4);
+    char h1 = hexlify(c & 0x0f);
+    tgt[0] = h0;
+    tgt[1] = h1;
+}
+
+
+int _get_fnr_flen(char* returned_value, uint64_t *fnr, uint32_t *flen){
+    printf("_get_fnr_flen\n");fflush(stdout);
+    //
+    // 03 3f 57 45 d9 02 00 00 00 00 00 00 00 00 d7 f2 0a 00
+    int pos = 0;
+    printf("xxxx\n");fflush(stdout);
+    uint8 checksum_type = returned_value[pos++];
+    printf("checksum_type:%i\n", checksum_type);fflush(stdout);
+    assert(checksum_type = 3);
```

Comment:
does this mean rora won't work when a different checksum than crc32c is configured?